crml/nft/src/lib.rs

```diff
@@ -853,6 +857,35 @@ decl_module! {
 				None => {},
 			}
 		}
+
+		/// Update fixed price for a single token sale
+		///
+		/// `listing_id` id of the fixed price listing
+		/// `new_price` new fixed price
+		/// Caller must be the token owner
+		#[weight = T::WeightInfo::update_fixed_price()]
+		fn update_fixed_price (
+			origin,
+			listing_id: ListingId,
+			new_price: Balance
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+
+			match Self::listings(listing_id) {
+				Some(Listing::<T>::FixedPrice(mut sale)) => {
+					ensure!(sale.seller == origin, Error::<T>::NoPermission);
+
+					sale.fixed_price = new_price;
+					let collection_id = sale.tokens[0].0;
+
+					Listings::mutate(listing_id, |listing| *listing = Some(Listing::<T>::FixedPrice(sale)));
```

Comment:
I believe you can just use an insert here
```suggestion
					Listings::insert(listing_id, Some(Listing::<T>::FixedPrice(sale)));
```

Reply:
yes `Listings::insert(listing_id, sale);` is better here since the value `(sale)` is already loaded.
`mutate` will try read the value from storage again and decode for no reason since it gets overwritten